README.md

```diff
@@ -1,2 +1,215 @@
-# openid-connect-bundle
-Symfony bundle for open-id connect
+# OpenId Connect Bundle
+
+Symfony bundle for authorization via OpenID Connect.
+
+## Installation
+
+To install run
+
+```shell
+composer require itk-dev/openid-connect-bundle
+```
+
+If you wish to run the coding standard tests for Markdown files
+
+```sh
+yarn install
+```
+
+## Usage
+
+Before being able to use the bundle,
+you must have your own User entity and database setup.
+
+Once you have this, you need to configure variables for
+OpenId Connect and create an Authenticator class that extends
+the bundle authenticator, `OpenIdLoginAuthenticator`.
+
+### Variable configuration
+
+In `/config/packages/` you need the following `itk_dev_openid_connect.yaml`
+file for configuring OpenId Connect variables
+
+```yaml
+itk_dev_open_id_connect:
+  open_id_provider_options:
+    urlConfiguration: 'https://.../openid-configuration..' // url to OpenId Discovery document
+    clientId: 'client_id' // Client id assigned by authorizer
+    clientSecret: 'client_secret' // Client password assigned by authorizer
+    cachePath: '' // Path for caching discovery document
+  open_id_return_route: 'some_route_name' // Route to where the user is to be redirected after success
```

Comment:
Maybe it should be clearer that this is the callback uri registered at the identity provider.
Also it could be that we shouldn't use the router, lets talk about this.